### PR TITLE
perf(ricos-editor): limit initial content change calls

### PR DESCRIPTION
**Pull request by oded-soffrin**
Due to performance issues, slack convo https://wix.slack.com/archives/C02Q305P07Q/p1639151433032100

other than boolean value reverse (@Barackos tell me if I missed something, but isn't it the _poet meaning_?)
this PR goal is to have a single expensive `getContentTraits` call

### CODE REVIEW

**Comment by Barackos** [comment]
@oded-soffrin Oops 😄 yes the value reverse is definitely right. Should've respected the poet a little more

**Comment by oded-soffrin** [comment]
> @oded-soffrin Oops 😄 yes the value reverse is definitely right. Should've respected the poet a little more

And regarding saving extra calls, seems right?

**Comment by Barackos** [comment]
> And regarding saving extra calls, seems right?

Yes, in fact - following your change, you can also remove the `if` clause inside `onInitialContentChanged`, as it is always true

File: packages/ricos-editor/web/src/RicosEditor.tsx [hunk 1]
```diff
@@ -225,17 +225,18 @@ export class RicosEditor extends Component<RicosEditorProps, State> {
 
   onInitialContentChanged = () => {
     const { initialContentChanged } = this.state;
```
Comment (Barackos) [inline packages/ricos-editor/web/src/RicosEditor.tsx:227]:
```suggestion
```

File: packages/ricos-editor/web/src/RicosEditor.tsx [hunk 2]
```diff
@@ -225,17 +225,18 @@ export class RicosEditor extends Component<RicosEditorProps, State> {
...
-    if (initialContentChanged) {
+    if (!initialContentChanged) {
       const contentId = this.getContentID();
       this.getBiCallback('onContentEdited')?.({ version: Version.currentVersion, contentId });
```
Comment (Barackos) [inline packages/ricos-editor/web/src/RicosEditor.tsx:230]:
```suggestion
    const contentId = this.getContentID();
    this.getBiCallback('onContentEdited')?.({ version: Version.currentVersion, contentId });
```

File: packages/ricos-editor/web/src/RicosEditor.tsx [hunk 3]
```diff
@@ -225,17 +225,18 @@ export class RicosEditor extends Component<RicosEditorProps, State> {
...
-      this.setState({ initialContentChanged: false });
+      this.setState({ initialContentChanged: true });
     }
```
Comment (Barackos) [inline packages/ricos-editor/web/src/RicosEditor.tsx:232]:
```suggestion
    this.setState({ initialContentChanged: true });
```

**Review by Barackos — APPROVED** [review]

File: packages/ricos-editor/web/src/RicosEditor.tsx (hunk 1 above)
Comment (Barackos) [reply packages/ricos-editor/web/src/RicosEditor.tsx:227]: Sorry for the 3 code suggestions - Github doesn't allow 1 suggestion when some lines are "deleted" 😄

**Comment by lxgreen** [comment]
@Barackos are you sure it is necessary to store the `initialContentChanged` on state? Could be just a class property to save re-renders, right?

**Comment by Barackos** [comment]
> @lxgreen: Are you sure it is necessary to store the `initialContentChanged` on state? Could be just a class property to save re-renders, right?

True, moved to class variable

**Pull request merged**